### DISCUSSION TITLE
Redact Sensitive Info by default

### DISCUSSION THREAD

**Discussion by TLMondo** [directus/directus#19130] [Feature Requests]
### Summary

Certain data points by standard practice should be redacted (in some cases, regardless of Role). It would be nice to have a standard set documented that will be redacted in all UI/API/Logs/etc., and then have individual config settings to override that based on `"always"` or `"by role"` or `"under conditions"`. This way the strictest level of protection is default, and someone like me dumping the `{{payload}}` attribute to see what I want to use in my flow body template, is safe from logging people's passwords.

Suggests sensitive variable names:

- password
- credit card
- social security

Then if I am working on an App or Function that needs the `credit card` value, I can put some restrictive rules that make exceptions.

Ideally, we could define a whole list of variable names and/or regex patterns, and each time something is displayed/saved, it passes through a sanitation process first. Less efficient yes, but much less hassle than a lawsuit.

### Motivation

Trying to ensure that future admins and developers have a safety-first enforcement layer, but the mechanism to override.

### Detailed Design

See above

### Must Have:
- `password` redacted in all uses

### Should Have:
- A means to configure default-to-redacted variables to have exceptions

### Could Have:
- A configuration section and parameters to set a custom list for redaction through standardized sanitation functions

### Won't Have:
- 

### Drawbacks

- Performance hit

### Alternatives

I believe that there should never be an exception to `password`, and users should reset password if having issues, vs sharing.

### Adoption Strategy

- Release Notes to call out new, potentially breaking, behavior. 
- Active affirmation of change when accessing the admin panel, per user, after update

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
